Sources/LiveKit/Core/Engine.swift

```diff
@@ -755,6 +755,50 @@ extension Engine: ConnectivityListenerDelegate {
 
 // MARK: Engine - Factory methods
 
+private extension Array where Element: RTCVideoCodecInfo {
+
+    func swapH264Codec() -> [RTCVideoCodecInfo] {
+        // 42e032
+        guard let profileLevelId = RTCH264ProfileLevelId(profile: .constrainedBaseline, level: .level5) else {
+            // this should never happen
+            logger.log("failed to generate profileLevelId", .error, type: Engine.self)
+            return self
+        }
+
+        // create a new H264 codec with new profileLevelId
+        let newH264 = RTCVideoCodecInfo(name: kRTCH264CodecName,
+                                        parameters: ["profile-level-id": profileLevelId.hexString,
+                                                     "level-asymmetry-allowed": "1",
+                                                     "packetization-mode": "1"])
+
+        // swap the h264 codec
+        let codecs = map { $0.name == kRTCVideoCodecH264Name ? newH264 : $0 }
```

Comment:
Does this line means replace all h264 codec profile to 42e032? is this change will affect iOS devices?

Reply:
if it is, we might need some device compatible test

Reply:
Yes it will replace all for iOS/macOS